### PR TITLE
feat: alert on scheduled delivery error to slack notification channel

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #8520 

### Description:

Allows adding `notification_channel` to the Slack App integration

When that exists, and there's an error on a Scheduled Delivery, that will send a message to that notification channel, like so: 
image

If the user goes back to clearing the channel (on the integrations page) it will skip this feature.

Demo:

https://github.com/lightdash/lightdash/assets/7611706/508c591d-ad7f-4040-acc8-9f68cb67b9f6

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/SlackSettingsPanel/index.tsx [hunk 1]
```diff
@@ -25,66 +40,130 @@ export const hasRequiredScopes = (slackSettings: SlackSettings) => {
         slackRequiredScopes.length
     );
 };
+
+const SLACK_INSTALL_URL = `/api/v1/slack/install/`;
```
Comment (ZeRego) [inline packages/frontend/src/components/UserSettings/SlackSettingsPanel/index.tsx:44]: good constant 👍

File: packages/backend/src/controllers/slackController.ts [hunk 2]
```diff
@@ -41,4 +44,28 @@ export class SlackController extends Controller {
             results: await slackClient.getChannels(organizationUuid),
         };
     }
+
+    /**
+     * Update slack notification channel to send notifications to scheduled jobs fail
+     * @param req express request
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
```
Comment (ZeRego) [inline packages/backend/src/controllers/slackController.ts:52]:
```suggestion
    @Middlewares([allowApiKeyAuthentication, isAuthenticated, unauthorisedInDemo])
```

File: packages/backend/src/clients/Slack/SlackClient.ts [hunk 3]
```diff
@@ -214,9 +214,86 @@ export class SlackClient {
...
+            organizationUuid,
+            channelId,
+        );
+
+        if (channelId) {
+            await this.slackApp.client.chat
+                .postMessage({
+                    token: installation?.token,
+                    channel: channelId,
+                    text: 'This channel will now receive notifications when scheduled jobs fail',
```
Comment (ZeRego) [inline packages/backend/src/clients/Slack/SlackClient.ts:252]: Would be nice to have more context. No mention of Lightdash and who did it.

File: packages/backend/src/clients/Slack/SlackMessageBlocks.ts [hunk 4]
```diff
@@ -309,3 +309,32 @@ export const getUnfurlBlocks = (
                   }),
     },
 });
+
+export const getNotificationChannelErrorBlocks = (
+    schedulerName: string,
+    error: any,
+): KnownBlock[] =>
+    getBlocks([
```
Comment (ZeRego) [inline packages/backend/src/clients/Slack/SlackMessageBlocks.ts:317]: Would be nice to have a link to the scheduler logs.

**Review by ZeRego — APPROVED** [review]

**Review by ZeRego — APPROVED** [review]
good job 🥇 🚀

File: packages/backend/src/clients/Slack/SlackMessageBlocks.ts (hunk 4 above)
Comment (joaoviana) [reply packages/backend/src/clients/Slack/SlackMessageBlocks.ts:317]: sure, I think this can be done separately if that's ok

**Comment by joaoviana** [comment]
e2e test error only on invites.cy.ts - unrelated so will merge

**Pull request merged**